### PR TITLE
📖 Link architecture and code walkthrough videos in docs 

### PR DESCRIPTION
**Pull request by mrajashree**
**What this PR does / why we need it**: This PR adds links to some of the videos available online that explain cluster api architecture or include code walkthrough of some components.

### CODE REVIEW

File: docs/book/src/developer/guide.md [hunk 1]
```diff
@@ -221,3 +221,12 @@ Now you can [create CAPI objects][qs]!
 To test another iteration, you'll need to follow the steps to build, push, update the manifests, and apply.
 
 [qs]: https://cluster-api.sigs.k8s.io/user/quick-start.html#usage
+
+## Videos explaining CAPI architecture and code walkthrough
+
+- CAPI components and architecture
+	- https://youtu.be/npFO5Fixqcc
```
Comment (CecileRobertMichon) [inline docs/book/src/developer/guide.md:228]:
```suggestion
	- [Cluster API Deep Dive](https://youtu.be/npFO5Fixqcc)
```

Consider formatting the links with the title of the video so it's easier to know which one is which when we add more to the list in the future

Comment (CecileRobertMichon) [inline docs/book/src/developer/guide.md:228]: Consider formatting the links with the title of the video so it's easier to know which one is which when we add more to the list in the future

```suggestion
	- [Cluster API Deep Dive](https://youtu.be/npFO5Fixqcc)
```

Comment (vincepri) [reply docs/book/src/developer/guide.md:228]: Let's also maybe add the year and target version to these?

Comment (mrajashree) [reply docs/book/src/developer/guide.md]: @vincepri I added in the year and month of the videos, kind of hard to tell the target version from the videos though. is year/month enough? For the CAPD walkthrough I've added v1alpha4

File: docs/book/src/developer/guide.md [hunk 2]
```diff
@@ -221,3 +221,15 @@ Now you can [create CAPI objects][qs]!
 To test another iteration, you'll need to follow the steps to build, push, update the manifests, and apply.
 
 [qs]: https://cluster-api.sigs.k8s.io/user/quick-start.html#usage
+
+## Videos explaining CAPI architecture and code walkthrough
+
+- CAPI components and architecture
+
+	- [Cluster API Deep Dive - Dec 2020](https://youtu.be/npFO5Fixqcc)
```
Comment (sbueringer) [inline docs/book/src/developer/guide.md:229]: I think there's a tab vs. spaces problem :) (same in l.235)

Comment (sbueringer) [inline docs/book/src/developer/guide.md:229]: This is v1alpha3

File: docs/book/src/developer/guide.md [hunk 3]
```diff
@@ -221,3 +221,15 @@ Now you can [create CAPI objects][qs]!
...
 
 [qs]: https://cluster-api.sigs.k8s.io/user/quick-start.html#usage
+
+## Videos explaining CAPI architecture and code walkthrough
+
+- CAPI components and architecture
+
+	- [Cluster API Deep Dive - Dec 2020](https://youtu.be/npFO5Fixqcc) 
+  - [Cluster API Deep Dive - Sept 2020](https://youtu.be/9SfuQQeeK6Q)
+  - [Declarative Kubernetes Clusters with Cluster API - Oct 2020](https://youtu.be/i6OWn2zRsZg)
```
Comment (sbueringer) [inline docs/book/src/developer/guide.md:231]: also v1alpha3

**Comment by sbueringer** [comment]
> not sure why the indentation is a bit off in the diff, if I click on "view file" it shows the right indentation.

one tab vs 2 spaces :)

File: docs/book/src/developer/guide.md (hunk 1 above)
Comment (ashish-amarnath) [reply docs/book/src/developer/guide.md]: should we include other kubecon videos as well?

File: docs/book/src/developer/guide.md (hunk 2 above)
Comment (mrajashree) [reply docs/book/src/developer/guide.md]: great thanks!

File: docs/book/src/developer/guide.md (hunk 1 above)
Comment (mrajashree) [reply docs/book/src/developer/guide.md]: sure, I tried adding only the latest videos, but if there's any other specific videos you want to add let me know 👍

Comment (ashish-amarnath) [reply docs/book/src/developer/guide.md]: I could find these

https://www.youtube.com/watch?v=AeHfVFepsMg
https://www.youtube.com/watch?v=9SfuQQeeK6Q
https://www.youtube.com/watch?v=B_Zb9_Ralb4

Also, including the latest makes sense. I'll let others chime in on whether to add these or now.

Comment (mrajashree) [reply docs/book/src/developer/guide.md]: ok thanks, the second one's already in the PR, I can add in the remaining two

Comment (mrajashree) [reply docs/book/src/developer/guide.md]: @ashish-amarnath the first video is v1alpha3 version right? and what about the third?

**Comment by sbueringer** [comment]
/retest
we have to many flakes :)

**Comment by CecileRobertMichon** [comment]
/retest

related to #4466

**Comment by sbueringer** [comment]
/retest

All unrelated flakes, as in this PR only doc was changed. Shouldn't need a lot more retries after #4466 has been merged

**Pull request merged**